### PR TITLE
Fixed issue where country field didn't work correctly outside of a form

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
#62

### CODE REVIEW

File: src/app/visual/country-field/country-field-visual.component.ts [hunk 1]
```diff
@@ -52,4 +52,8 @@ export class CountryFieldVisualComponent implements OnInit {
       this.disableFields = true;
     }
   }
+
+  public countryChanged(country: any) {
+    console.log('Change occurred: ' + country);
```
Comment (jeffbdye) [inline src/app/visual/country-field/country-field-visual.component.ts]: Very minor, but is it alright to log in test code?

Comment (Blackbaud-TrevorBurch) [reply src/app/visual/country-field/country-field-visual.component.ts]: Good catch. Didn't mean to check in the visual test changes. I've stripped that back out. The actual testing is done in the unit tests. I was just using this for local testing.

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-country-field-non-form](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-country-field-non-form)

File: src/app/public/modules/country-field/fixtures/country-field.component.fixture.ts [hunk 2]
```diff
@@ -28,4 +28,8 @@ export class CountryFieldTestComponent {
 
   public modelValue: SkyCountryFieldCountry;
 
+  public countryChanged(country: SkyCountryFieldCountry) {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/country-field/fixtures/country-field.component.fixture.ts]: Missing a return type ☝️

File: src/app/public/modules/country-field/fixtures/country-field-reactive.component.fixture.ts [hunk 3]
```diff
@@ -75,4 +75,8 @@ export class CountryFieldReactiveTestComponent implements OnInit {
     });
   }
 
+  public countryChanged(country: SkyCountryFieldCountry) {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/country-field/fixtures/country-field-reactive.component.fixture.ts]: Missing a return type.

File: src/app/public/modules/country-field/fixtures/country-field-no-form.component.fixture.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  @ViewChild(SkyCountryFieldComponent)
+  public countryFieldComponent: SkyCountryFieldComponent;
+
+  public defaultCountry: string;
+
+  public isDisabled: boolean = false;
+
+  public isRequired: boolean = false;
+
+  public countryChanged(country: SkyCountryFieldCountry) {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/country-field/fixtures/country-field-no-form.component.fixture.ts]: Missing a return type.

File: src/app/public/modules/country-field/fixtures/country-field-no-form.component.fixture.html [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<div>
+  <sky-country-field
+    [disabled]="isDisabled ? true : undefined"
+    [defaultCountry]="defaultCountry"
+    [attr.required]="isRequired ? true : undefined"
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/country-field/fixtures/country-field-no-form.component.fixture.html]: Check your alphabetical ordering of properties here.

Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/country-field/fixtures/country-field-no-form.component.fixture.html]: Done

File: src/app/public/modules/country-field/fixtures/country-field-no-form.component.fixture.ts (hunk 4 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/country-field/fixtures/country-field-no-form.component.fixture.ts]: Done

File: src/app/public/modules/country-field/fixtures/country-field-reactive.component.fixture.ts (hunk 3 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/country-field/fixtures/country-field-reactive.component.fixture.ts]: Done

File: src/app/public/modules/country-field/fixtures/country-field.component.fixture.ts (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/country-field/fixtures/country-field.component.fixture.ts]: Done

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-country-field-non-form](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-country-field-non-form)

File: src/app/public/modules/country-field/country-field.component.ts [hunk 6]
```diff
@@ -87,7 +73,21 @@ export class SkyCountryFieldComponent implements ControlValueAccessor, OnDestroy
    * Indicates whether to disable the country field.
    */
   @Input()
-  public disabled: boolean = false;
+  public set disabled(isDisabled: boolean) {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/country-field/country-field.component.ts]: I noticed toggling the disabled state will reset the values for pristine and touched. For example, load the form then edit one of the inputs so it is no longer pristine. Then, click the button to disable the forms and you'll notice pristine gets reset back to `true` and touched back to `false`. I tried this with a basic reactive input and did not have the same results. Is it supposed to do this?

Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/country-field/country-field.component.ts]: 👀

Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/country-field/country-field.component.ts]: I just added a normal text input inside the form and see the same behavior where if I edit that input and then disable that input the outer form reverts to pristine?

Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/country-field/country-field.component.ts]: Happy to hop on a call with you if it is easier but I'm seeing the same behavior with the basic input as with the country field.

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/country-field/country-field.component.ts]: Ok, I double checked this and it turns out I had a nested form group that was causing issues. You're right - this is the expected Angular behavior. Thanks for checking!

**Pull request merged**